pkg/encoding/kzgEncoder/multiframe.go

```diff
@@ -0,0 +1,171 @@
+package kzgEncoder
+
+import (
+	"bytes"
+	"encoding/gob"
+	"errors"
+	"fmt"
+
+	rs "github.com/Layr-Labs/eigenda/pkg/encoding/encoder"
+	bls "github.com/Layr-Labs/eigenda/pkg/kzg/bn254"
+)
+
+// Sample is the basic unit for a verification
+// A blob may contain multiple Samples
+type Sample struct {
+	Commitment bls.G1Point
+	Proof      bls.G1Point
+	Row        int // corresponds to a row in the verification matrix
+	Coeffs     []bls.Fr
+	X          uint // X is assignment
+}
+
+// generate a random value using Fiat Shamir transform
+// we can also pseudo randomness generated locally, but we have to ensure no adv can manipulate it
+// Hashing everything takes about 1ms, so Fiat Shamir transform does not incur much cost
+func GenRandomness(params rs.EncodingParams, samples []Sample, m int) (bls.Fr, error) {
+	var buffer bytes.Buffer
+	enc := gob.NewEncoder(&buffer)
+
+	for _, sample := range samples {
+		err := enc.Encode(sample.Commitment)
+		if err != nil {
+			return bls.ZERO, err
+		}
+	}
+
+	var randomFr bls.Fr
+
+	err := bls.HashToSingleField(&randomFr, buffer.Bytes())
```

Comment:
Does it need to add a random salt to actually make it random (now it's a deterministic hash)?

Reply:
It is gist of fiat-Shamir transform https://en.wikipedia.org/wiki/Fiat%E2%80%93Shamir_heuristic

Reply:
No, it does not need to be deterministic.

Reply:
I haven't read details about Fiat Shamir, but intuitively, if this "randomness" is actually deterministic and determined by the input samples, which is supplied by the prover, can the verifier be fooled by the prover?
It looks the verifier needs some randomness that the prover can never guess.